pkg/fakerp/admin_handlers.go

```diff
@@ -1,58 +1,40 @@
 package fakerp
 
 import (
-	"encoding/json"
 	"net/http"
 
 	"github.com/go-chi/chi"
 
 	"github.com/openshift/openshift-azure/pkg/api"
 )
 
-func (s *Server) adminreply(w http.ResponseWriter, err error, out interface{}) {
-	if err != nil {
-		s.internalError(w, err.Error())
-		return
-	}
-
-	if out == nil {
-		return
-	}
-
-	if b, ok := out.([]byte); ok {
-		w.Header().Set("Content-Type", "application/octet-stream")
-		w.Write(b)
-		return
-	}
-
-	b, err := json.Marshal(out)
-	if err != nil {
-		s.internalError(w, err.Error())
-		return
-	}
-	w.Header().Set("Content-Type", "application/json")
-	w.Write(b)
-	return
-}
-
 // handleBackup handles admin requests to backup an etcd cluster
 func (s *Server) handleBackup(w http.ResponseWriter, req *http.Request) {
-	cs := s.read()
-	if cs == nil {
+	cs, err := s.store.Get()
```

Comment:
every handler needs a cs.  Every put handler writes a cs.  DRY.  Try to move the get and optional put logic into context() in middleware.go - it's mad to have it duplicated in every handle function.

Reply:
This adds some ambiguity in the error handling when we need to know if this is first Put create, or update Put, etc. We cant error on non-existing record/file because of case 1.